### PR TITLE
[5.2] Update namespace during make:auth

### LINKED ISSUE #11847: php artisan auth:make command does not use the customer create app:name Namespace
Hi

Created a new project in laravel 5.2, changed the namespace using the php artisan app:name xyz

when running the php artisan auth:make command it still uses the default laravel namespace "app" instead of XYZ that was

Please fix this for future

### PR DESCRIPTION
**Pull request by azeemhassni**
When you change App name using `app:name Acme` the new namespace will reflect on the controller generated using `make:auth`.

---

Closes #11847.

### CODE REVIEW

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 1]
```diff
@@ -26,14 +26,14 @@ class MakeAuthCommand extends Command
      * @var array
      */
     protected $views = [
-        'auth/login.stub' => 'auth/login.blade.php',
-        'auth/register.stub' => 'auth/register.blade.php',
+        'auth/login.stub'           => 'auth/login.blade.php',
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: Please revert these changes.

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 2]
```diff
@@ -47,19 +47,24 @@ public function fire()
 
         $this->exportViews();
 
-        if (! $this->option('views')) {
+        if (!$this->option('views')) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: revert this

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 3]
```diff
@@ -47,19 +47,24 @@ public function fire()
...
             $this->info('Installed HomeController.');
 
-            copy(
-                __DIR__.'/stubs/make/controllers/HomeController.stub',
-                app_path('Http/Controllers/HomeController.php')
+            $homeControllerStub = file_get_contents(__DIR__ . '/stubs/make/controllers/HomeController.stub');
+
+            $this->replaceNamespace($homeControllerStub)
+                ->replaceImports($homeControllerStub);
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: this is just fine on one line pleae

**Comment by Francismori7** [comment]
You broke styling guidelines. See https://laravel.com/docs/5.2/contributions#coding-style

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 4]
```diff
@@ -47,19 +47,24 @@ public function fire()
...
+                app_path('Http/Controllers/HomeController.php'),
+                $homeControllerStub
             );
 
             $this->info('Updated Routes File.');
 
             file_put_contents(
                 app_path('Http/routes.php'),
-                file_get_contents(__DIR__.'/stubs/make/routes.stub'),
+                file_get_contents(__DIR__ . '/stubs/make/routes.stub'),
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: Please stop making these changes

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 5]
```diff
@@ -74,15 +79,15 @@ public function fire()
      */
     protected function createDirectories()
     {
-        if (! is_dir(base_path('resources/views/layouts'))) {
+        if (!is_dir(base_path('resources/views/layouts'))) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: Please stop making these changes

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 6]
```diff
@@ -95,11 +100,40 @@ protected function createDirectories()
...
+            'App\Http\Controllers', $this->laravel->getNamespace() . 'Http\Controllers', $stub
+        );
+
+        return $this;
+    }
+
+    /**
+     * Replace namespace in the list list of imports for the given stub
+     *
+     * @param $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: invalid phpdoc

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 7]
```diff
@@ -95,11 +100,40 @@ protected function createDirectories()
...
 
-            copy(__DIR__.'/stubs/make/views/'.$key, $path);
+            copy(__DIR__ . '/stubs/make/views/' . $key, $path);
         }
     }
+
+    /**
+     * Replace the namespace for the given stub.
+     *
+     * @param  string $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: cs

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 8]
```diff
@@ -95,11 +100,40 @@ protected function createDirectories()
...
+     *
+     * @param $stub
+     * @return $this
+     */
+    protected function replaceImports( &$stub )
+    {
+        $stub = str_replace(
+            'App\Http\Requests', $this->laravel->getNamespace() . 'Http\Requests', $stub
+        );
+        return $this;
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: cs

**Comment by GrahamCampbell** [comment]
Please squash to one commit.

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 9]
```diff
@@ -102,4 +107,34 @@ protected function exportViews()
             copy(__DIR__.'/stubs/make/views/'.$key, $path);
         }
     }
+
+    /**
+     * Replace the namespace for the given stub.
+     *
+     * @param  string $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: cs

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 10]
```diff
@@ -102,4 +107,34 @@ protected function exportViews()
...
+            'App\Http\Controllers', $this->laravel->getNamespace().'Http\Controllers', $stub
+        );
+
+        return $this;
+    }
+
+    /**
+     * Replace namespace in the list of imports for the given stub.
+     *
+     * @param $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: invalid phpdoc

File: src/Illuminate/Auth/Console/MakeAuthCommand.php [hunk 11]
```diff
@@ -102,4 +107,34 @@ protected function exportViews()
...
+     * @return $this
+     */
+    protected function replaceImports(&$stub)
+    {
+        $stub = str_replace(
+            'App\Http\Requests', $this->laravel->getNamespace().'Http\Requests', $stub
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/Console/MakeAuthCommand.php]: What if the namespace wasn't App, say on a 2nd run of the command.

**Comment by taylorotwell** [comment]
Holding off on this for now.
